traffic_ops/traffic_ops_golang/cachegroup/queueupdate.go

```diff
@@ -49,10 +49,14 @@ func QueueUpdates(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	if reqObj.CDN == nil && reqObj.CDNID == nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("cdn does not exist"), nil)
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("cdn is a required field"), nil)
 		return
 	}
-	if reqObj.CDN == nil || *reqObj.CDN == "" {
+	if *reqObj.CDN == "" {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("cdn cannot be blank"), nil)
+		return
+	}
+	if reqObj.CDNID != nil && reqObj.CDN == nil || *reqObj.CDN == "" {
```

Comment:
`&&` has higher precedence than `||` in Go, so it binds first. This is equivalent to `if (reqObj.CDNID != nil && reqObj.CDN == nil) || *reqObj.CDN == "" {`. Which panics on the dereference immediately below if the CDNID is nil, and `CDN` is not nil and the empty string. 

Needs to be `if reqObj.CDNID != nil && (reqObj.CDN == nil || *reqObj.CDN == "") { `.

Even if `&&` and `||` had equal precedence evaluated from the left, IMO it's still always better to parenthesize different logical operators, to make it obviously correct.

See: https://golang.org/ref/spec#Operator_precedence
Example: https://play.golang.org/p/L3FkgKISgFZ